syntax.php

```diff
@@ -56,6 +52,12 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
                 }
             }
         }
+        // Set default flag values
+        if (!$flags) {
+            $flags = [
+                'showCount' => false,
+            ];
+        }
```

Comment:
Sorry, using line numbers was confusing, this what I tried to say. Thanks for the effort.
```Diff
if (preg_match ('/\[.*\]/', $junk, $flags) === 1) {
             $flags = trim ($flags [0], '[]');
             $found = explode(',', $flags);
-            $flags = array();
+             $flags = [
+                 'showCount' => false,
+             ];
            foreach ($found as $flag) {
                if (in_array($flag, $this->knownFlags)) {
                    // Actually we just set flags as present
                    // Later we might add values to flags like key=value pairs
                    $flags [$flag] = true;
                 }
             }
         }
-         // Set default flag values
-       if (!$flags) {
-             $flags = [
-                 'showCount' => false,
-             ];
-         }
         


```

Reply:
Now that I removed the extra check at line 145 per our earlier discussion, initializing $flags before the foreach as you suggest would expose us to another undefined array key warning, if preg_match() didn't match.

If your intention is to ensure the code is forward-compatible with future new flags, then I would propose to initialize the $flags array before the `if (preg_match` (like I did in the previous commit), and use another variable for preg_match() matches.

```php
$flags = [
    'showCount' => false,
];
if (preg_match('/\[.*\]/', $junk, $matches) === 1) {
    $matches = trim($matches[0], '[]');
    $found = explode(',', $matches);
    $flags = array();
    foreach ($found as $flag) {
        if (in_array($flag, $this->knownFlags)) {
            // Actually we just set flags as present
            // Later we might add values to flags like key=value pairs
            $flags [$flag] = true;
        }
    }
}
```

Let me know what you think, before I commit.

Reply:
Ah, sure! I agree

Reply:
OK, change has been committed.